packages/shared/apis/EthereumController.ts

```diff
@@ -49,26 +50,38 @@ export interface IEthereumController {
 export class EthereumController extends ExposableAPI implements IEthereumController {
   @exposeMethod
   async requirePayment(toAddress: string, amount: number, currency: string): Promise<any> {
+    await this.ensureHasPermissions()
     return requirePayment(toAddress, amount, currency)
   }
 
   @exposeMethod
   async signMessage(message: MessageDict) {
+    await this.ensureHasPermissions()
     return signMessage(message)
   }
 
   @exposeMethod
   async convertMessageToObject(message: string): Promise<MessageDict> {
+    await this.ensureHasPermissions()
     return convertMessageToObject(message)
   }
 
   @exposeMethod
   async sendAsync(message: RPCSendableMessage): Promise<any> {
+    await this.ensureHasPermissions()
     return sendAsync(message)
   }
 
   @exposeMethod
   async getUserAccount(): Promise<string | undefined> {
+    await this.ensureHasPermissions()
     return getUserAccount(requestManager)
   }
+
+  private async ensureHasPermissions() {
```

Comment:
could this fn be at the `ExposableAPI` ? So we dont need to re-write it on everey api we create.
Also it could receive an array of permissions.
```ts
private async ensureHasPermissions(permisions: keyofPermissions[])
```

Reply:
Maybe we can create another ExposableAPI like RestrictedExposableAPI to give this default methods?